pkg/controller/registry/reconciler/grpc.go

```diff
@@ -478,9 +486,23 @@ func (c *GrpcRegistryReconciler) removePods(pods []*corev1.Pod, namespace string
 // CheckRegistryServer returns true if the given CatalogSource is considered healthy; false otherwise.
 func (c *GrpcRegistryReconciler) CheckRegistryServer(catalogSource *v1alpha1.CatalogSource) (healthy bool, err error) {
 	source := grpcCatalogSourceDecorator{CatalogSource: catalogSource, createPodAsUser: c.createPodAsUser, opmImage: c.opmImage, utilImage: c.utilImage}
+
+	// The CheckRegistryServer function is called by the CatalogSoruce controller before the registry resources are created,
+	// returning a IsNotFound error will cause the controller to exit and never create the resources, so we should
+	// only return an error if it is something other than a NotFound error.
+	serviceAccount := source.ServiceAccount()
+	serviceAccount, getError := c.OpClient.GetServiceAccount(serviceAccount.GetNamespace(), serviceAccount.GetName())
+	if getError != nil {
+		healthy = false
+		if !apierrors.IsNotFound(getError) {
+			err = getError
+		}
+		return
```

Comment:
Naked named returns are one of Go's silliest features - given that the `healthy` variable only ever exists to return, would be OK in this PR changing this function to do a more normal flow?


```go
if getError != nil {
    if apierrors.IsNotFound(getError) {
        return false, nil
    }
    return false, getError
}